qtservice/src/qtservice_unix.cpp

```diff
@@ -191,7 +231,23 @@ QString QtServiceController::serviceFilePath() const
 
 bool QtServiceController::uninstall()
 {
-    QSettings settings(QSettings::SystemScope, "QtSoftware");
+
+    if (isSystemD()) {
+        uninstallSysD(systemDPath() + serviceName() + ".service");
+    }
+
+    return uninstallUpStart();
+}
+
+bool QtServiceController::uninstallSysD(const QString &serviceFile)
+{
+    return QFile::remove(serviceFile);
+
+}
+
+bool QtServiceController::uninstallUpStart()
+{
+    auto &settings = getSettings(serviceName());
```

Comment:
There are not modify. use QSettings settings(QSettings::SystemScope, "QtSoftware");